### PR TITLE
Ignore foreign keys in tables definition

### PR DESCRIPTION
**Pull request by jsc0218**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...Ignore foreign keys in tables definition to improve compatibility with MySQL, so a user wouldn't need to rewrite his SQL of the foreign key part, ref https://github.com/ClickHouse/ClickHouse/issues/53380

### Documentation entry for user-facing changes
- Motivation:  previously, MySQL users have to remove their foreign key declaration in their SQL script to submit to ClickHouse, because ClickHouse would treat foreign key declaration as illegal. With this feature, foreign keys would be recognized and ignored, so no necessary to modify MySQL script of the foreign key part. 

- Example use: following queries can be executed now with foreign key omit:
-- CREATE TABLE child  (id INT, pid INT, PRIMARY KEY(id), FOREIGN KEY(pid) REFERENCES parent(pid)) ENGINE MergeTree;
-- CREATE TABLE child (id INT, pid INT, PRIMARY KEY(id), FOREIGN KEY(pid) REFERENCES parent(pid) ON DELETE CASCADE ON UPDATE RESTRICT) ENGINE MergeTree;

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit bf12415215e1d95ec03bf615a6ffdfbfde9e8b2f with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53864/bf12415215e1d95ec03bf615a6ffdfbfde9e8b2f/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🔴 failure</td></tr>
</table>

**Review by alexey-milovidov — APPROVED** [review]
LGTM

File: src/Parsers/ParserCreateQuery.cpp [hunk 1]
```diff
@@ -222,17 +223,69 @@ bool ParserProjectionDeclaration::parseImpl(Pos & pos, ASTPtr & node, Expected &
     return true;
 }
 
+bool ParserForeignKeyDeclaration::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
```
Comment (tavplubix) [inline src/Parsers/ParserCreateQuery.cpp:226]: We already have (almost) the same code: https://github.com/ClickHouse/ClickHouse/blob/1d170f57457fd575714797615b65f2304746e06c/src/Parsers/MySQL/ASTDeclareReference.cpp#L46-L102

Let's reuse it and avoid code duplication.

**Review by tavplubix — COMMENTED** [review]
I'm not sure that it's okay to silently ignore it by default. Maybe we should introduce a setting that allows ignoring it, and throw an exception if it's disabled

**Comment by jsc0218** [comment]
Thanks for pointing out the similar code there. It's a good question to discuss. I think it depends on the context whether foreign key should be ignored in all engines or only MySQL engine.

1. The current implementation favours ignoring fk in all engines. If this is expected, it would be better to flesh out the code in the main parser, and then the code in MySQL parser dir references the main parser code.
2. If ignoring fk is only in MySQL engine, not in other engines, the code would be better to reside in MySQL parser dir and gets referenced in main parser.

Since it would be very costly to support fk in merge tree engine as well as the ANSI compatibility in the doc, I would assume it is the case 1. Either approach would require clean the code. Anyway, I am quite new to ClickHouse, please correct my thought.

**Pull request merged**